Neos.ContentGraph.DoctrineDbalAdapter/src/DoctrineDbalContentGraphProjection.php

```diff
@@ -237,13 +230,7 @@ public function getState(): ContentGraphFinder
 
     public function markStale(): void
     {
-        $contentGraphFinder = $this->getState();
-        /** @var ContentGraph $contentGraph */
-        foreach ($contentGraphFinder->getInstances() as $contentGraph) {
-            foreach ($contentGraph->getSubgraphs() as $subgraph) {
-                $subgraph->inMemoryCache->disable();
-            }
-        }
+        $this->getState()->forgetInstances();
```

Comment:
Ok this specifically now breaks the one failing test, but it's amazing that of all the tests we just have one failing because this here is a fundamental break. 

The `\Neos\ContentRepository\Core\EventStore\EventPersister::publishEvents()` does call markStale() but that means the contentStreamIdOverrides are gone as well so during a workspace publish/rebase operation some events might apply to the wrong content stream as the overriden "special" instance was purged and will then automatically be replaced by a automatically resolved (by workspace) one which has the wrong content stream.

We _could_ separate the override instances but that seems messy as well?

Reply:
Wow very good catch. This is indeed a flaw that should be guarded against.

> We _could_ separate the override instances but that seems messy as well?

i would actually prefer some `$this->isInOverrideState` flag than how you implemented it sneakily ^^